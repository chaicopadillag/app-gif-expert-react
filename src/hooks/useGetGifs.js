import { useState } from 'react';
import { getGifs } from '../api/getGifs';

export const useGetGifs = (categoria) => {
	const [estadoGifs, setEstadoGifs] = useState({
		data: [],
		loading: true,
	});
	getGifs(categoria).then((gifs) => {
		setEstadoGifs({
			data: gifs,
			loading: false,
		});
	});
	return estadoGifs;
};
