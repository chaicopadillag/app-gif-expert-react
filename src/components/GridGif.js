import React from 'react';
import PropTypes from 'prop-types';
import { useGetGifs } from '../hooks/useGetGifs';
import ItemGifGrit from './ItemGifGrit';

const GridGif = ({ categoria }) => {
	const { data: gifs, loading } = useGetGifs(categoria);
	return (
		<>
			<h2 className="animate__animated animate__fadeIn"> {categoria}</h2>
			<p className="animate__animated animate__flash">{loading ? 'Cargando...' : 'Cargado!!!'}</p>
			<div className="card-grid">
				{gifs.map((gif) => (
					<ItemGifGrit key={gif.id} {...gif} />
				))}
			</div>
		</>
	);
};
GridGif.propTypes = {
	categoria: PropTypes.string.isRequired,
};

export default GridGif;
