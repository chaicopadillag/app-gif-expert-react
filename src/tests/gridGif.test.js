import React from 'react';
import { shallow } from 'enzyme';
import GridGif from '../components/GridGif';
import { useGetGifs } from '../hooks/useGetGifs';

jest.mock('../hooks/useGetGifs');

describe('Prueba de Componente <GridGif/>', () => {
	const categoria = 'Terror';

	useGetGifs.mockReturnValue({
		data: [],
		loading: true,
	});
	const wraper = shallow(<GridGif categoria={categoria} />);

	test('Debe hace to matchSnapshot', () => {
		expect(wraper).toMatchSnapshot();
	});

	test('Simular que haya cargados imagenes useGetImagenes', () => {
		const data = [
			{
				id: '123',
				title: 'Titulo de la IMG',
				url: 'https://localhost/img/foto.jpg',
			},
			{
				id: 'ACB',
				title: 'Titulo de la AB',
				url: 'https://localhost/img/foto2.jpg',
			},
		];
		useGetGifs.mockReturnValue({
			data: data,
			loading: false,
		});
		const wraper = shallow(<GridGif categoria={categoria} />);
		expect(wraper).toMatchSnapshot();
		expect(wraper.find('p').exists()).toBe(true);
		expect(wraper.find('ItemGifGrit').length).toBe(data.length);
	});
});
