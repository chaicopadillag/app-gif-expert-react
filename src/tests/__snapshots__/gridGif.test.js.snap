// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de Componente <GridGif/> Debe hace to matchSnapshot 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
     
    Terror
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba de Componente <GridGif/> Simular que haya cargados imagenes useGetImagenes 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
     
    Terror
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Cargado!!!
  </p>
  <div
    className="card-grid"
  >
    <ItemGifGrit
      id="123"
      key="123"
      title="Titulo de la IMG"
      url="https://localhost/img/foto.jpg"
    />
    <ItemGifGrit
      id="ACB"
      key="ACB"
      title="Titulo de la AB"
      url="https://localhost/img/foto2.jpg"
    />
  </div>
</Fragment>
`;
